### PR TITLE
fix(buildDockerAndPublishImage) annotated tags

### PR DESCRIPTION
**Pull request by lemeurherve**
Fixes https://github.com/jenkins-infra/helpdesk/issues/2887 by annotating the tag so it's an object itself with its date, and not just a reference attached to the last commit

### CODE REVIEW

File: vars/buildDockerAndPublishImage.groovy [hunk 1]
```diff
@@ -138,7 +138,7 @@ def call(String imageName, Map userConfig=[:]) {
             usernamePassword(credentialsId: "${finalConfig.gitCredentials}", passwordVariable: 'GIT_PASSWORD', usernameVariable: 'GIT_USERNAME')
           ]) {
             echo "Tagging New Version: $nextVersion"
-            sh "git tag $nextVersion"
+            sh "git tag -a $nextVersion -m $IMAGE_NAME"
```
Comment (dduportal) [inline vars/buildDockerAndPublishImage.groovy:141]: Good idea to add a commit message with the image name!

But you have to also update the tests (check for `-a` flag + you have to mock the $IMAGE_NAME` variable in the sh mock)

File: test/groovy/BuildDockerAndPublishImageStepTests.groovy [hunk 2]
```diff
@@ -44,6 +44,7 @@ class BuildDockerAndPublishImageStepTests extends BaseTest {
     helper.addShMock('git remote -v | grep origin | grep push | sed \'s/^origin\\s//\' | sed \'s/\\s(push)//\'', 'https://github.com/org/repository.git', 0)
     helper.addShMock('gh api /repos/org/repository/releases | jq -e -r \'.[] | select(.draft == true and .name == "next") | .id\'', '12345', 0)
     addEnvVar('WORKSPACE', '/tmp')
+    addEnvVar('IMAGE_NAME', testImageName)
```
Comment (lemeurherve) [inline test/groovy/BuildDockerAndPublishImageStepTests.groovy:47]:
```suggestion
```

**Comment by lemeurherve** [comment]
Restarted another build, all good :)
image

**Review by dduportal — APPROVED** [review]
LGTM!

File: vars/buildDockerAndPublishImage.groovy [hunk 3]
```diff
@@ -137,8 +137,12 @@ def call(String imageName, Map userConfig=[:]) {
           withCredentials([
             usernamePassword(credentialsId: "${finalConfig.gitCredentials}", passwordVariable: 'GIT_PASSWORD', usernameVariable: 'GIT_USERNAME')
           ]) {
+            echo "Configuring git user and email"
+            sh 'git config user.name "jenkins"'
```
Comment (dduportal) [inline vars/buildDockerAndPublishImage.groovy:141]:
```suggestion
            sh 'git config user.name "${GIT_USERNAME}"'
```
WDYT?

Comment (lemeurherve) [reply vars/buildDockerAndPublishImage.groovy:141]: Not sure it's appropriate, these tags can be unrelated to any commiter.

**Pull request merged**